src/lib/webpack/render-html-plugin.js

```diff
@@ -0,0 +1,44 @@
+import { resolve } from 'path';
+import { existsSync } from 'fs';
+import HtmlWebpackExcludeAssetsPlugin from 'html-webpack-exclude-assets-plugin';
+import ScriptExtHtmlWebpackPlugin from 'script-ext-html-webpack-plugin';
+import HtmlWebpackPlugin from 'html-webpack-plugin';
+import { readJson } from './webpack-base-config';
+import prerender from './prerender';
+
+export default function (config) {
+	const { cwd, dest, isProd, src } = config;
+
+	const htmlWebpackConfig = ({ url, title }) => ({
+		filename: resolve(dest, url.substring(1), 'index.html'),
+		template: `!!ejs-loader!${config.template || resolve(__dirname, '../../resources/template.html')}`,
+		minify: isProd && {
+			collapseWhitespace: true,
+			removeScriptTypeAttributes: true,
+			removeRedundantAttributes: true,
+			removeStyleLinkTypeAttributes: true,
+			removeComments: true
+		},
+		favicon: existsSync(resolve(src, 'assets/favicon.ico')) ? 'assets/favicon.ico' : resolve(__dirname, '../../resources/favicon.ico'),
+		manifest: config.manifest,
+		inject: true,
+		compile: true,
+		preload: config.preload===true,
+		title: title || config.title || config.manifest.name || config.manifest.short_name || (config.pkg.name || '').replace(/^@[a-z]\//, '') || 'Preact App',
+		excludeAssets: [/(bundle|polyfills)(\..*)?\.js$/],
+		config,
+		ssr(params) {
+			return config.prerender ? prerender({ cwd, dest, src }, { ...params, url }) : '';
+		}
+	});
+
+	const pages = readJson(resolve(cwd, config.prerenderUrls || '')) || [{ url: '/' }];
```

Comment:
Should't this be using `params.url` for the fallback?

Reply:
For `pages`? The `params` is only within the `ssr` function. I think the `/` route is a safe default. Plus, it's just copied over from what was there already.

Reply:
Yea maybe we should do it in a separate PR. Basically, the whole "subdirectory" stuff is something we should fix. But yea, let's keep it separate.